src/index.js

```diff
@@ -1,34 +1,48 @@
 import React from "react";
 import ReactDOM from "react-dom/client";
+import { useState } from "react";
+import SearchComponent from "./components/search-component/index";
 import HeaderComponent from "./components/header-component";
 import CardComponent from "./components/card-component/index.js";
 import teamdata from "./components/card-component/config";
+import "./index.scss";
+
+// Team card component (assignment-4) and search functionality (assignment-5)
 
-const CardContainer = () => {
-  // using for loop to iterate array of team data objects
-  // let card = [];
-  // for (let i = 0; i < teamdata.length; i++) {
-  //   console.log(i);
-  //   console.log(teamdata[i]);
-  //   card.push(<CardComponent teamdata={teamdata[i]} key={teamdata.id} />);
-  // }
-  const card = teamdata.map((element) => {
-    return <CardComponent teamdata={element} key={element.id} />;
-  });
-  return card;
-};
 const BodyComponent = () => {
+  const [searchfiltertext, setsearchfiltertext] = useState(teamdata);
   return (
     <div className="card__body">
       <h2 style={{ textAlign: "center" }}>
         Here's our team. <br></br> Meet our Team Members
       </h2>
+      <SearchComponent setsearchfiltertext={setsearchfiltertext} />
       <div className="card__containers">
-        <CardContainer />
+        <CardContainer searchfiltertext={searchfiltertext} />
       </div>
     </div>
   );
 };
+const CardContainer = ({ searchfiltertext }) => {
+  // using for loop to iterate array of team data objects
+  // let card = [];
+  // for (let i = 0; i < teamdata.length; i++) {
+  //   console.log(i);
+  //   console.log(teamdata[i]);
+  //   card.push(<CardComponent teamdata={teamdata[i]} key={teamdata.id} />);
+  // }
+  console.log(searchfiltertext);
+  const searchingresult = () => {
+    return searchfiltertext.map((element) => {
+      return <CardComponent teamdata={element} key={element.id} />;
```

Comment:
If that's for playing around then it's okay. But you should put your component logic inside/after the return keyword only, e.g., here- in return we are calling a function that returns an array of JSX elements. Doesn't one have to think to read it?... If one person needs to think means you should make it more readable.